core-blocks/file/edit.js

```diff
@@ -0,0 +1,252 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { getBlobByURL, revokeBlobURL } from '@wordpress/utils';
+import {
+	ClipboardButton,
+	IconButton,
+	Toolbar,
+	withNotices,
+} from '@wordpress/components';
+import { Component, compose, Fragment } from '@wordpress/element';
+import {
+	MediaUpload,
+	MediaPlaceholder,
+	BlockControls,
+	RichText,
+	editorMediaUpload,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+import FileBlockInspector from './inspector';
+import FileBlockEditableLink from './editable-link';
+
+class FileEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		const {
+			href,
+			showDownloadButton = true,
+			buttonText = __( 'Download' ),
+			id,
+		} = this.props.attributes;
+
+		this.onSelectFile = this.onSelectFile.bind( this );
+
+		// Initialize default values if undefined
+		this.props.setAttributes( {
+			showDownloadButton,
+			buttonText,
+		} );
+
+		// edit component has its own attributes in the state so it can be edited
+		// without setting the actual values outside of the edit UI
+		this.state = {
+			showPlaceholder: ! href,
+			href,
+			attachmentPage: undefined,
+			showCopyConfirmation: false,
+		};
+
+		if ( id !== undefined ) {
+			// Get Attachment Page URL
+			wp.apiRequest( {
+				path: `/wp/v2/media/${ id }`,
+				method: 'GET',
+			} )
+				.then(
+					( result ) => {
+						this.setState( { attachmentPage: result.link } );
+					},
+					() => {
+						this.setState( { attachmentPage: undefined } );
+					}
+				);
+		}
+	}
+
+	componentDidMount() {
+		const { href } = this.state;
+
+		// Upload a file drag-and-dropped into the editor
+		if ( this.isBlobURL( href ) ) {
+			getBlobByURL( href )
+				.then( ( file ) => {
+					editorMediaUpload( {
+						allowedType: '*',
```

Comment:
Just to make sure I understand correctly, once #6968 gets in we will feed the list if types here?

Reply:
From #5462:
> WordPress also supports a number of filetypes that aren't media, such as pdf, doc, and others. See full list here: https://codex.wordpress.org/Uploading_Files

I don't think that `editorMediaUpload` in the current shape is what we really want to use in here, because non-media files can be uploaded, too. It seems like we need to refactor the existing method or add a more generic one which `editorMediaUpload` uses behind the scenes.

Reply:
Hmm, I was under the impression that the "media" in `editorMediaUpload` and `mediaUpload` is being used in the broader sense of the word, and not to mean "images, audio, and video". This is in line with how WordPress uses the term — every kind of uploadable file, including pdf and other documents, are uploaded to the "Media Library", and are retrieved via the `media` endpoint.

Even [/edit-post/hooks/blocks/media-upload](https://github.com/WordPress/gutenberg/blob/master/edit-post/hooks/blocks/media-upload/index.js), which includes specific logic for when gallery == true, seems generic enough for any kind of file. Is there a reason we want to keep `editorMediaUpload` specifically for image/audio/video files?

Reply:
I posted a separate issue for the `allowedType` problem: #7155